File: app/src/main/java/com/lazylee/apiguidedemo/intents/IntentsFragment.java
package com.lazylee.apiguidedemo.intents;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.lazylee.apiguidedemo.R;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link IntentsFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class IntentsFragment extends Fragment {
    public static final String TAG = "IntentsFragment";




    public IntentsFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @return A new instance of fragment IntentsFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static IntentsFragment newInstance() {
        IntentsFragment fragment = new IntentsFragment();

        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.intents_fragment, container, false);
    }

}

File: app/src/main/java/com/lazylee/apiguidedemo/coretopics/intents/IntentsFragment.kt
package com.lazylee.apiguidedemo.coretopics.intents

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.lazylee.apiguidedemo.R

class IntentsFragment : Fragment(), View.OnClickListener {
    private var mBtn: Button? = null
    private var mBtn2: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val rootView = inflater.inflate(R.layout.intents_fragment, container, false)
        mBtn = rootView.findViewById(R.id.button)
        mBtn2 = rootView.findViewById(R.id.common_intent)
        mBtn?.setOnClickListener(this)
        mBtn2?.setOnClickListener(this)
        return rootView
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.button -> {
            }
            R.id.common_intent -> {
                val intent = Intent(context, CommonIntentActivity::class.java)
                startActivity(intent)
            }
            else -> {
            }
        }
    }

    companion object {
        const val TAG = "IntentsFragment"

        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @return A new instance of fragment IntentsFragment.
         */
        fun newInstance(): IntentsFragment {
            return IntentsFragment()
        }
    }
}